protean/version.hpp

```diff
@@ -6,22 +6,16 @@
 #ifndef PROTEAN_VERSION_HPP
 #define PROTEAN_VERSION_HPP
 
-#define PROTEAN_MAJOR_VERSION 1
-#define PROTEAN_MINOR_VERSION 8
+#define PROTEAN_MAJOR_VERSION 2
+#define PROTEAN_MINOR_VERSION 21
 #define PROTEAN_PATCH_VERSION 0
 
-//
-//  PROTEAN_VERSION % 100 is the patch level
-//  PROTEAN_VERSION / 100 % 1000 is the minor version
-//  PROTEAN_VERSION / 100000 is the major version
+#define PROTEAN_VERSION ((PROTEAN_MAJOR_VERSION * 100000) + (PROTEAN_MINOR_VERSION * 1000) + PROTEAN_PATCH_VERSION)
 
-#define PROTEAN_VERSION 100800
-
-//
 //  PROTEAN_LIB_VERSION must be defined to be the same as PROTEAN_VERSION
 //  but as a *string* in the form "x_y[_z]" where x is the major version
 //  number, y is the minor version number, and z is the patch level if not 0.
 
-#define PROTEAN_LIB_VERSION "1_08"
+#define PROTEAN_LIB_VERSION "2_21"

```

Comment:
It's yonks since I did any C++, but can't this be composed from the version defines above?

Reply:
Indeed, but only with more messing around and pulling in of extra Boost headers than I could be bothered with